### ISSUE TITLE
’not found data‘erro when training

### ISSUE THREAD

**Issue by kaelem1** [bmaltais/kohya_ss#2428]
17:45:46-693860 INFO     Training has ended.
17:52:57-749330 INFO     Start training LoRA Standard ...
17:52:57-751289 INFO     Validating lr scheduler arguments...
17:52:57-751289 INFO     Validating optimizer arguments...
17:52:57-752288 INFO     Validating model file or folder path
                         G:/Comfyui/ComfyUI-aki-v1.3/models/checkpoints/sdXL_v10.safetensors existence...
17:52:57-754288 INFO     ...valid
17:52:57-754288 INFO     Validating output_dir path G:/Comfyui/kohyass/output existence...
17:52:57-755710 INFO     ...valid
17:52:57-756715 INFO     Validating train_data_dir path G:/Comfyui/kohyass/train data/30_2D existence...
17:52:57-757716 INFO     ...valid
17:52:57-757716 INFO     reg_data_dir not specified, skipping validation
17:52:57-759719 INFO     logging_dir not specified, skipping validation
17:52:57-762722 INFO     log_tracker_config not specified, skipping validation
17:52:57-763719 INFO     resume not specified, skipping validation
17:52:57-764718 INFO     vae not specified, skipping validation
17:52:57-765720 INFO     network_weights not specified, skipping validation
17:52:57-766717 INFO     dataset_config not specified, skipping validation
17:52:57-769718 INFO     Regulatization factor: 1
17:52:57-771717 INFO     Total steps: 0
17:52:57-772717 INFO     Train batch size: 2
17:52:57-773738 INFO     Gradient accumulation steps: 1
17:52:57-774817 INFO     Epoch: 30
17:52:57-775824 INFO     Max train steps: 1600
17:52:57-777824 INFO     stop_text_encoder_training = 0
17:52:57-777824 INFO     lr_warmup_steps = 0
17:52:57-780823 INFO     Saving training config to G:/Comfyui/kohyass/output\glass_20240501-175257.json...
17:52:57-782823 INFO     Executing command: "G:\Comfyui\kohyass\kohya_ss\venv\Scripts\accelerate.EXE" launch
                         --dynamo_backend no --dynamo_mode default --mixed_precision bf16 --num_processes 1
                         --num_machines 1 --num_cpu_threads_per_process 2
                         "G:/Comfyui/kohyass/kohya_ss/sd-scripts/sdxl_train_network.py" --config_file
                         "./outputs/config_lora-20240501-175257.toml" with shell=True
17:52:57-788823 INFO     Command executed.
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: No module named 'triton'
2024-05-01 17:53:04 INFO     Loading settings from ./outputs/config_lora-20240501-175257.toml...      train_util.py:3744
                    INFO     ./outputs/config_lora-20240501-175257                                    train_util.py:3763
2024-05-01 17:53:04 INFO     prepare tokenizers                                                   sdxl_train_util.py:134
2024-05-01 17:53:06 INFO     update token length: 75                                              sdxl_train_util.py:159
                    INFO     Using DreamBooth method.                                               train_network.py:172
                    INFO     prepare images.                                                          train_util.py:1572
                    INFO     0 train images with repeating.                                           train_util.py:1613
                    INFO     0 reg images.                                                            train_util.py:1616
                    WARNING  no regularization images / 正則化画像が見つかりませんでした              train_util.py:1621
                    INFO     [Dataset 0]                                                              config_util.py:565
                               batch_size: 2
                               resolution: (1024, 1024)
                               enable_bucket: False
                               network_multiplier: 1.0

                    INFO     [Dataset 0]                                                              config_util.py:571
                    INFO     loading image sizes.                                                      train_util.py:853
0it [00:00, ?it/s]
                    INFO     make buckets                                                              train_util.py:859
                    WARNING  min_bucket_reso and max_bucket_reso are ignored if bucket_no_upscale is   train_util.py:876
                             set, because bucket reso is defined by image size automatically /
                             bucket_no_upscaleが指定された場合は、bucketの解像度は画像サイズから自動計
                             算されるため、min_bucket_resoとmax_bucket_resoは無視されます
                    INFO     number of images (including repeats) /                                    train_util.py:905
                             各bucketの画像枚数（繰り返し回数を含む）
G:\Comfyui\kohyass\kohya_ss\venv\lib\site-packages\numpy\core\fromnumeric.py:3504: RuntimeWarning: Mean of empty slice.
  return _methods._mean(a, axis=axis, dtype=dtype,
G:\Comfyui\kohyass\kohya_ss\venv\lib\site-packages\numpy\core\_methods.py:129: RuntimeWarning: invalid value encountered in scalar divide
  ret = ret.dtype.type(ret / rcount)
                    INFO     mean ar error (without repeats): nan                                      train_util.py:915
                    ERROR    No data found. Please verify arguments (train_data_dir must be the     train_network.py:212
                             parent of folders with images) /
                             画像がありません。引数指定を確認してください（train_data_dirには画像が
                             あるフォルダではなく、画像があるフォルダの親フォルダを指定する必要があ
                             ります）
17:53:08-090420 INFO     Training has ended.

**Comment by kaelem1** [comment]
my image folder name：30_2D

**Comment by bmaltais** [comment]
You need to put the parent folder of that firectory as the image directory... not the actual folder containing the images. So if `30_2D` is in a folder called `img` then put `img` as the image folder.

**Comment by bmaltais** [comment]
> *This was generated by AI during triage.*

## Duplicate

Closing as a **duplicate of #3374**.

### Why this is the same issue
The `No data found` / `train_data_dir must be the parent of folders with images` error almost always means one of:

1. **Image folder** points at the leaf folder that holds the images (`…/10_concept`) instead of the **parent** of the `N_name` subfolders (`…/img`).
2. Subfolders are not named with the required `N_concept` pattern (e.g. `goth` instead of `10_goth`).

Required layout (DreamBooth / LoRA / TI folder mode):

```text
train_data_dir/          ← set this path in the GUI
  10_my_concept/
    image1.jpg
    image1.txt
```

Docs: [`docs/image_folder_structure.md`](https://github.com/bmaltais/kohya_ss/blob/master/docs/image_folder_structure.md)

### What happens next
#3374 is kept open as the **canonical** ticket for a product fix: the GUI already has a `verify_image_folder_pattern` helper but it is never called on train start, so users only see the opaque error from sd-scripts. Tracking the implementation work there.
